src/libcore/str/mod.rs

```diff
@@ -3554,6 +3554,7 @@ impl str {
     ///
     /// assert_eq!("Hello\tworld", s.trim());
     /// ```
+    #[must_use]
```

Comment:
I think these deserve a "reason" argument (which gets tucked into a note on the diagnostic display). Compare:

https://github.com/rust-lang/rust/blob/a72a0801bdab22d7c0167f67d1148276ea874d2e/src/liballoc/str.rs#L210-L211

Reply:
Thanks, I added a message.